dart/constraint/ConstraintSolver.cpp

```diff
@@ -515,6 +520,10 @@ void ConstraintSolver::updateConstraints()
     }
     else
     {
+      // Increment the count of contacts between the two collision objects
+      ++contactPairMap[std::make_pair(
+          contact.collisionObject1, contact.collisionObject2)];
```

Comment:
does this assume that the ordering of collision objects is consistent, i.e. that for any contacts between shape `A` and shape `B`, that `A` will always be `collisionObject1` and `B` collisionObject2` or vice versa?

Reply:
Good point. That was my assumption. I added a comparison function that ensures that the order doesn't matter. https://github.com/azeey/dart/pull/9/commits/8e173fbb3199da890e0e14a7ac98c2004ef66fc7

Reply:
just checking that I'm reading it correctly: are they sorted by their pointer address?

Reply:
Yes

Reply:
To be clear, my original assumption was that the ordering was consistent. The change in https://github.com/azeey/dart/commit/8e173fbb3199da890e0e14a7ac98c2004ef66fc7 doesn't make that assumption anymore.